File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int maxDepth(TreeNode* root) {
        int n1=1,n2=1;
        if (root==NULL) {
            return 0;
        }
        n1=maxDepth(root->left)+n1;
        n2=maxDepth(root->right)+n2;
        
        if (n1>n2) {
            return n1;
        }
        else {
            return n2;
        }
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        count = 0

        def recurse(node, maxseen):

            # 
            if node == None:
                return 0

            l = 0
            if node.val >= maxseen:

                maxseen = node.val
                l+=1

            return l + recurse(node.left,maxseen) + recurse(node.right,maxseen) 

        return recurse(root,-101)            